### PR TITLE
Add automatically prioritising todo extension

### PR DESCRIPTION
**Pull request by ThomasPinna**
## Description

This [Raycast](https://raycast.com/) extension helps you manage your to-dos by automatically prioritising them for you.

*How it works*

When you create a new to-do, you can mark it with any of the following flags.

- _Important_: This to-do is relatively important compared to other tasks.
- _Quick_: This to-do can be completed with relatively low effort.
- _Urgent_: This to-do has a deadline, and it feels relatively close by.

Based on these flags a score is assigned to these to-dos. The basic formula is as follows:

```python
todo.urgency * 3 + todo.importance * 2 + todo.quickness * 1.1
```

Your to-dos will then be automatically ordered based on this score. When a todo is completed, then it will appear in a
separate section in order to not clutter the list of to-dos that still need to be performed

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/todo/src/todo.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+  return (todo) => query_matchers.every((matcher) => matcher(todo));
+}
+
+// helper functions, no mutations
+
+export function priority_score(todo: Todo) {
+  let score = 0;
+  if (todo.quick) score += 1.1;
+  if (todo.important) score += 2;
```
Comment (pernielsentikaer) [inline extensions/todo/src/todo.ts:56]:
``` suggestion
  else if (todo.important) score += 2;
```

File: extensions/todo/src/todo.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  if (todo.urgent) score += 3;
```
Comment (pernielsentikaer) [inline extensions/todo/src/todo.ts:57]:
```suggestion
  else if (todo.urgent) score += 3;
```

File: extensions/todo/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+"semi": true,
+"tabWidth": 2,
+"printWidth": 100,
+"singleQuote": true,
+"trailingComma": "all",
+"jsxSingleQuote": true,
+"bracketSpacing": true
+}
```
Comment (pernielsentikaer) [inline extensions/todo/.prettierrc:9]: {
  "printWidth": 120,
  "singleQuote": false
}

File: extensions/todo/src/backup-actions.component.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { Action, confirmAlert, showHUD } from '@raycast/api';
+import { BACKUP_PATH } from './constants';
+import { load_todos_from_selected_file, restore_todos } from './backup';
+import { Todo } from './todo';
+
+export function ShowBackupsAction() {
+  return (
+    <Action.ShowInFinder
+      path={BACKUP_PATH}
+      title='Show Backups in finder'
```
Comment (pernielsentikaer) [inline extensions/todo/src/backup-actions.component.tsx:10]:
```suggestion
      title='Show Backups in Finder'
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
- I would call it Todo instead of To-do 🙂

- Here it should show e.g. `Mark Quick`

Raycast-2LHFMenh@2x

- The colors in accessory icons are hard to see on dark background

todo 2023-06-12 at 11 22 15

**Comment by ThomasPinna** [comment]
I had a discussion with my wife about Todo vs to-do, she argued that to-do is actually the more correct choice. A quick google resulted in this thread: [word choice - "Todo list" or "to-do list" - English Language & Usage Stack Exchange](https://english.stackexchange.com/questions/46217/todo-list-or-to-do-list#:~:text=You%20can%20use%20whichever%20you,off%20sticking%20to%20to%2Ddo.)


About the colours, do you have any suggestions? Or are you talking about the grey colours, because these are hard to see by design, since they are "inactive".

File: extensions/todo/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "root": true,
+  "env": {
+    "es2020": true,
+    "node": true
+  },
+  "parser": "@typescript-eslint/parser",
+  "plugins": ["@typescript-eslint"],
+  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended", "prettier"]
+}
```
Comment (thomaslombart) [inline extensions/todo/.eslintrc.json:10]: You can use `npx ray migrate` to make use of our latest ESLint config.

File: extensions/todo/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+import { to_backup } from './backup';
+import { RestoreBackupAction } from './backup-actions.component';
+
+export default function Command() {
+  // todos
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [dones, setDones] = useState<Todo[]>([]);
+
+  // the app is loading when loading todos
+  const [loading, setLoading] = useState<boolean>(true);
```
Comment (thomaslombart) [inline extensions/todo/src/index.tsx:16]: Note that you can use `useCachedPromise` from `@raycast/utils` for something like that.

**Review by thomaslombart — CHANGES_REQUESTED** [review]
Congrats on the extension! I have some feedback:

- Could you find a differentiating name for your extension. `todo` sounds too generic and we already have a simple to-do list extension in the store?
- Could you add a screencast of your extension?
- Could you [adjust your icon](https://developers.raycast.com/basics/prepare-an-extension-for-store#extension-icon) so that there's no extra whitespace on the borders?
- Could you title-case your actions' titles as well as adding colors to the different icons?
- Could you use symbols? (`⌘` instead of `cmd`)

**Pull request closed without merging**